### ISSUE TITLE
Again: Tab content and titles disrupted

### ISSUE THREAD

**Issue by martinstoehr** [Sequel-Ace/Sequel-Ace#1797]
- Sequel Ace Version (including build number): 4.0.6 (20048)
- Sequel Ace Source (App Store/GitHub/Homebrew): App Store
- macOS Version: 13.5
- Processor Type (Intel/Apple): Apple
- MySQL Version: all
- macOS Localization: en, de

With the latest version all the serious errors from the past are here again... 

Please have a look at this: https://github.com/Sequel-Ace/Sequel-Ace/issues/1056

In addition Sequel Ace locks up when having more than two tabs open with ssh-connection and closing one of them.

**Comment by HJCunha** [comment]
I'm seeing this issue happening too. It was there in the past, got solved at some point but now is back and is annoyingly dangerous as we believe we are editing one server but we can be changing a different one..

**Comment by anthonyvdg-se** [comment]
Seems to happen often when creating new tabs while there are other active tabs

**Comment by xixi773** [comment]
Hello,

I can confirm the issue with:

- Sequel Ace Version (including build number): 4.0.7 (20050)
- Sequel Ace Source (App Store/GitHub/Homebrew): App Store
- macOS Version: 13.4
- Processor Type (Intel/Apple): Intel
- MySQL Version: all
- macOS Localization: en

**Comment by ArneAmeye** [comment]
I'm experiencing this too lately, on 4.0.6 I started noticing this, upgraded to  4.0.7 version and still seeing this bug.
Tabs get mixed up all the time, it's dangerous if you don't pay attention to it since you might be editing or export/import data that you did not expect to work with.

**Comment by cl2714** [comment]
Can you fix this bug as soon as possible? It's really too dangerous

**Comment by dlt-** [comment]
Same here. Happens with Mysql/mariadb connection tabs too. Shows another database tables/content (which is not even open anymore in any visible tab)

Sequel Ace Version (including build number): 4.0.7 (20050)
Sequel Ace Source (App Store/GitHub/Homebrew): App Store
macOS Version: 14.0beta
Processor Type (Intel/Apple): Apple
MySQL Version: all
macOS Localization: en

**Comment by stevierayrobison** [comment]
Same issue. Dangerous. Window title correct. Tab title incorrect.
![Screenshot 2023-08-23 at 4 44 54 PM](https://github.com/Sequel-Ace/Sequel-Ace/assets/23525587/bb994efd-1ff5-4523-ae9d-367c0b22b273)

Sequel Ace Version (including build number): 4.0.7 (20050)
Sequel Ace Source (App Store/GitHub/Homebrew): App Store
macOS Version: Ventura 13.4.1
Processor Type (Intel/Apple): Apple
MySQL Version: all (AWS RDS)
macOS Localization: en

**Comment by Jason-Morcos** [comment]
I am able to reproduce this but I can't figure out how to fix it. I'm hopeful that @Kaspik may be able to find a fix, but in the interim I encourage you to only open new tabs when the last tab is active. If anyone else out there is able to fix this, I would gladly review that PR.
The simple steps to reproduce are to open a new tab when the active tab _isn't_ the last tab in the tab bar.

**Comment by somegooser** [comment]
Hi,

I dont know how to fix this bug but i know the way to **reproduce** the bug all the time.

The problem appears when you pres the 'plus' sign to add a new tab while not being active on the last tab.

So example if you have 3 tabs open and currently active on the second tab. If you press add new tab the connection will be added to the LAST tab while the last tab is already a working connection. 

But if the third (last tab in this example) is active and you press add new tab it will work just fine.

This bug can easily be fixed if the new tab ALWAYS gets added as last.

**Comment by Mike-LA-UK** [comment]
Not sure about it being related to the plus button or if that's just a good way to replicate it. For me it basically seems that if I have three tabs open, let my mac sleep overnight, come back in the morning then there's a decent chance that the 2nd and 3rd tab have now swapped and will mismatch to what the actual database is.

It's super dangerous as far as bugs go for a database management client (I know everyone has said that, but it really is!).

Hadn't spotted it before because I used to use a different client for prod to dev/etc - will go back to that for safety. Hope someone can work out the cause as it's a fantastic client otherwise!

**Comment by dgm** [comment]
It bugged on my simply dragging the last tab over to reorder; the last two tabs became the same database as the original last tab, and I could not access what had been the next to last tab.

**Comment by Kaspik** [comment]
The only thing that changed between 4.0.5 and 4.0.6 and could be related is this PR: https://github.com/Sequel-Ace/Sequel-Ace/pull/1736 - although I'm not sure how it could.

Hopefully found a fix and open a PR: https://github.com/Sequel-Ace/Sequel-Ace/pull/1988

TLDR - hard to believe but opening a new tab via "+" vs opening it via CMD+T differs from system standpoint.
In case user hits "+" in the UI in tab bar - system automatically creates a tab for us and adds it to the tabGroup - there is no way to avoid it and no way to work around it - and it adds it where it wants and doesn't care about the order. We do care about order but system just puts the tab to different place.

From now on, these two actions will differ a little - when user hits "+" in the UI, it will create a new tab always at the end of the tab bar list. If user hits "CMD+T" (or similar action in menu), it will create tab following the currently active tab.

**Comment by somegooser** [comment]
@Jason-Morcos hi jason did you see this pull request? and could it fix this problem?

**Comment by Kaspik** [comment]
Jason is enjoying a two week holiday and will take care of it over the weekend once he gets back :)

**Comment by Jason-Morcos** [comment]
Sorry for the delay, thank you, @Kaspik! New Release Candidate pushed to GitHub now

https://github.com/Sequel-Ace/Sequel-Ace/releases
